opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -75,14 +75,18 @@ impl SdkMeter {
     {
         let validation_result = validate_instrument_config(builder.name.as_ref(), &builder.unit);
         if let Err(err) = validation_result {
-            global::handle_error(err);
+            otel_error!(
+                name: "SdkMeter.CreateCounter.InstrumentCreationFailed", 
+                meter_name = self.scope.name.as_ref(),
+                instrument_name = builder.name.as_ref(),
+                error = format!("Measurements from the instrument will be ignored. Reason: {}", err));
```

Comment:
If we are using dedicated events for each instrument, then we might as well be more specific in the message.

```suggestion
                error = format!("Measurements from the counter will be ignored. Reason: {}", err));
```